### PR TITLE
Fixed Transaction listing to show deposit date

### LINKED ISSUE #584: Display deposit dates on transaction list
Extracted from #582 

> Once a deposit is posted, we should also show the deposit date, rather than "Deposit details".

Ideally, we'd use cached deposits for this, so we can retrieve the dates in one query.

### PR DESCRIPTION
**Pull request by jessy-p**
Fixes #584 

#### Changes proposed in this Pull Request

All rows in the transaction listing showed "Deposit Details" in the deposit column, though estimated or actual deposit date was available.

- Used the 'available_on' field of the transaction to display the date in the deposit column,  instead of the non-existent 'date_available' field.
- Prefix "Estimated" to estimated deposit date.
- Removed "Pending" or "Deposit details" from the deposit date column, as all deposits will have a generated or actual deposit id and estimated or actual deposit date. In the exception case (should never happen), the column is left blank.
- Fixed tests

#### Testing instructions

1. Go to Payment >> Transactions.
2. Deposits that have actual deposit dates, should show the deposit date in the last column.
3. Deposits that are estimated, or pending should say Estimated + the date in the deposit column

Screenshot 2021-06-07 at 12 59 11 PM

-------------------
- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

**Comment by jessy-p** [comment]
@LevinMedia 
After this fix, as all transactions have a deposit date, either estimated or actual, all the rows show the date. When should we show "Deposit Details" and "Pending"? Should we differentiate between the transactions with an estimated Deposit Date and transactions with actual deposits?

**Comment by LevinMedia** [comment]
@jessy-p Thanks for tackling this issue! With this addition, we no longer need the "deposit details" string for the deposit column. Any deposits that are estimated, or pending should say Estimated + the date in the deposit column.  

Estimated June 4, 2021

The entire string should link to the deposit details screen which that specific transaction is a part of.

File: client/transactions/list/deposit.js [hunk 1]
```diff
@@ -26,13 +25,17 @@ const Deposit = ( { depositId, dateAvailable } ) => {
 			true // TODO Change call to gmdateI18n and remove this deprecated param once WP 5.4 support ends.
 		);
 
-	return depositId ? (
+	const estimated =
+		depositId && depositId.includes( 'wcpay_estimated_' )
+			? 'Estimated'
```
Comment (dmvrtx) [inline client/transactions/list/deposit.js:30]: `'Estimated'` should be a translated string here. Use `__` similar to the removed `'Pending'` indicator.

**Review by dmvrtx — CHANGES_REQUESTED** [review]
Overall looks good, just a minor change to use translations instead of hard-coded string.

**Review by dmvrtx — APPROVED** [review]
Works like a charm! 🚀

**Pull request merged**